frontend/src/scenes/dashboard/Dashboard.tsx

```diff
@@ -19,38 +22,60 @@ export const Dashboard = hot(_Dashboard)
 function _Dashboard({ id, shareToken }: Props): JSX.Element {
     return (
         <BindLogic logic={dashboardLogic} props={{ id: parseInt(id), shareToken }}>
-            <DashboardView id={id} shareToken={shareToken} />
+            <DashboardView />
         </BindLogic>
     )
 }
 
-function DashboardView({ id, shareToken }: Props): JSX.Element {
-    const { dashboard, itemsLoading, items } = useValues(dashboardLogic)
-    const { user } = useValues(userLogic)
+function DashboardView(): JSX.Element {
+    const { dashboard, itemsLoading, items, isOnSharedMode, lastRefreshed } = useValues(dashboardLogic)
     const { dashboardsLoading } = useValues(dashboardsModel)
+    const { updateAndRefreshDashboard, refreshAllDashboardItems } = useActions(dashboardLogic)
+
+    if (dashboardsLoading || itemsLoading) {
+        return <SceneLoading />
+    }
+
+    if (!dashboard) {
+        return (
+            <>
+                <p>Dashboard not found.</p>
+            </>
+        )
+    }
 
     return (
-        <div style={{ marginTop: 32 }}>
-            {!shareToken && <DashboardHeader />}
+        <div className="dashboard">
+            {!isOnSharedMode && <DashboardHeader />}
 
-            {dashboardsLoading ? (
-                <SceneLoading />
-            ) : !dashboard ? (
-                <>
-                    <p>A dashboard with the ID {id} was not found!</p>
-                    <HedgehogOverlay type="sad" />
-                </>
-            ) : items && items.length > 0 ? (
-                <DashboardItems inSharedMode={!!shareToken} />
-            ) : itemsLoading ? (
-                <SceneLoading />
-            ) : user?.team?.ingested_event ? (
+            {items && items.length ? (
+                <div>
+                    <div className="dashboard-items-actions">
+                        <div className="left-item">
+                            Last updated <b>{moment(lastRefreshed).fromNow() || 'a while ago'}</b>
```

Comment:
Could we try to do this without `moment`? Not a blocker, just thinking about https://github.com/PostHog/posthog/pull/3490

Reply:
I think we need to get that PR in to use `dayjs` instead of `moment` so will leave for now.